API-GUIDELINE.md

```diff
@@ -0,0 +1,252 @@
+             {
+               "quantity": 2,
+               "state": {
+                 "typeId": "state",
+                 "id": "7c2e2694-aefe-43d7-888e-6a99514caaca"
+               }
+             }
+           ],
+           "priceMode": "Platform",
+           "lineItemMode": "Standard",
+           "totalPrice": {
+             "type": "centPrecision",
+             "fractionDigits": 2,
+             "currencyCode": "EUR",
+             "centAmount": 5600
+           }
+         }
+       ],
+       "store": {
+         "typeId": "store",
+         "key": "test-key"
+       },
+       "cartState": "Active",
+       "totalPrice": {
+         "type": "centPrecision",
+         "fractionDigits": 2,
+         "currencyCode": "EUR",
+         "centAmount": 5600
+       },
+       "customLineItems": [],
+       "discountCodes": [],
+       "inventoryMode": "None",
+       "taxMode": "Platform",
+       "taxRoundingMode": "HalfEven",
+       "taxCalculationMode": "LineItemLevel",
+       "refusedGifts": [],
+       "origin": "Customer"
+     }
+
+##Naming convention
+
+###General Guidelines
+To use consistent names these are the rules to follow:
+
+ DO'S
+ - use nouns instead of verbs to expose resources
+ - use lowercase and separated by hyphens (-) for path segments like /shopping-lists
+ - use American English
+ - create each resource as unique resource name
+ - use the proper HTTP methods for API operations whenever possible
+ 
+ DON'TS
+ - DON'T use snake case (_) or other punctuation
+ - DON'T use file extensions
+ - DON'T use abbreviations
+ - DON'T use CRUD HTTP methods or verbs
+
+###Naming Properties
+The properties are identified in base of the type: 
+ - DateTime
+ - Boolean
+ - Number/Integer
+ - String
+ - Array
+ - Object
+
+####DateTime
+In general, define a DateTime property using a name ending with "At", then if there is a range of date use "From" and "To".
+
+Some examples:
+
+        returnAt
+        expirationAt
+        validFrom
+        validTo
+
+####Boolean
+Define a Boolean property using a name which includes "is" or "has" and the verb at past participle.
+
+Some examples:
+
+        isEmailVerified
+        hasStagedChanges
+        isPublished
+        isActive
+        
+####Number/Integer
+Define a Number/Integer property using names that can be quantified.
+
+Some examples:
+
+        availableQuantity
+        restockableInDays
+        version
+        weight
+        height
+        orderNumber
+        variantId
+        totalAmount
+      
+####String
+Define a String property using names that have to be explicable.
+
+Some examples:
+
+      orderEditResult
+      productName
+      productKey
+      productSku
+      discountCode
+         
+####Array
+Define an Array property using names ending with "s".
+
+Some examples:
+
+    discountCodes
+    refusedGifts
+    roles
+    allowedValues
+    
+####Object
+Define an Object property using "ResourceIdentifier" or "Draft" or "Reference" in the name.
```

Comment:
Below examples contradict this statement.

It's more that for objects there are not that much rules.

Draft is used for objects which are meant to be written

ResourceIdentifier for object which refer to other resource and are meant to be written

References for objects which refer to other resources and are only read